core/src/testils/proptest/strategy.rs

```diff
@@ -259,12 +260,18 @@ fn srh<E: Debug + Ipv6Packet>(
         map.u8(&field::sr_segments_left),
         map.u16(&field::sr_tag),
     )
-        .prop_map(move |(packet, segments, segments_left, tag)| {
+        .prop_perturb(move |(packet, segments, segments_left, tag), mut rng| {
             let mut packet = packet.push::<SegmentRouting<E>>().unwrap();
             packet.set_segments(&segments).unwrap();
-            packet.set_segments_left(segments_left);
             packet.set_tag(tag);
             packet.set_next_header(next_header);
+
+            let num_segments = segments.len() as u8;
+            if segments_left > num_segments {
```

Comment:
cool but I feel this is not a good behavior. it's caller's responsibility to make sure data passed in is sane. don't like functions that decide if your input is not good, I will replace it for you with something. if we want this, we should make a version where segments + segments_left are just completely randomly generated.